File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> num_set(nums.begin(), nums.end());
        int res = 0;

        for (int num : num_set) {
            if (!num_set.count(num - 1)) {
                int currNum = num;
                int streak = 1;
                while (num_set.count(currNum + 1)) {
                    currNum = currNum + 1;
                    streak++;
                }
                res = max(res, streak);
            }
        }
        return res;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> vis = new HashSet<>();
        for (int i : nums) {
            vis.add(i);
        }
        int res = 0;
        for (int i : vis) {
            if (!vis.contains(i-1)) {
                int curr = i;
                int count = 0;
                while (vis.contains(curr)) {
                    count++;
                    curr = curr + 1;
                }
                res = Math.max(res, count);
            }
        }
        return res;

    }
}